src/tools/data_handlers/analytics_handler.py

```diff
@@ -149,13 +185,13 @@ async def _poll_for_completion(
         self,
         endpoint_url: str,
         payload: Dict,
-        max_retries: int = 3,
-        poll_interval: float = 0.5,
+        max_retries: int = 5,
+        poll_interval: float = 1,
     ) -> Dict | str:
         """Poll the API until the request is completed or max retries exceeded."""
         for attempt in range(max_retries):
             logger.info(f"Polling attempt {attempt + 1}/{max_retries}")
-            await asyncio.sleep(poll_interval)
+            await asyncio.sleep(poll_interval * (attempt + 1))
```

Comment:
Since this is progressive now, can we do poll inverval 0.5? This will increase by 0.5 secons in each of the retries right?